src/libFLAC/CMakeLists.txt

```diff
@@ -10,14 +10,29 @@ check_include_file("sys/param.h" HAVE_SYS_PARAM_H)
 set(CMAKE_REQUIRED_LIBRARIES m)
 check_function_exists(lround HAVE_LROUND)
 
-if(CMAKE_SYSTEM_PROCESSOR MATCHES "[xX]86_64|(AMD|amd)64")
-    if(CMAKE_SIZEOF_VOID_P EQUAL 4)
```

Comment:
Is the issue CMAKE_SIZEOF_VOID_P not working? I'd rather continue seeking alternatives than introduce C to our CMake files.

Reply:
`CMAKE_SIZEOF_VOID_P` returns value of system, not target. `CMAKE_SYSTEM_PROCESSOR` is also useless. More info: [How to detect target architecture using CMake?
](https://stackoverflow.com/questions/11944060/how-to-detect-target-architecture-using-cmake), [https://gitlab.kitware.com/cmake/cmake/issues/15170](CMAKE_SYSTEM_PROCESSOR is broken on Windows).

For `CMAKE_SYSTEM_PROCESSOR` is `AMD64` when i build in Visual Studio + Ninja + MSVC ARM64 toolchain.

Reply:
According to the documentation, [CMAKE_SIZEOF_VOID_P](https://cmake.org/cmake/help/latest/variable/CMAKE_SIZEOF_VOID_P.html) has been returning the value of the target machine since CMake 3.2. As far as CMAKE_SYSTEM_PROCESSOR goes, I wouldn't call it useless, we can still work around that bug IMO. In the case that we hit AMD64, we can also check against CMAKE_CL_64 and if it is false, then set IA32. Google does a [similar](https://boringssl.googlesource.com/boringssl/+/refs/heads/master/CMakeLists.txt#472) check in BoringSSL.

Reply:
Updating Vcpkg package manager FLAC port. a've found `arm-windows` and `arm64-windows` builds both failed.

I've tryed all known checks: `CMAKE_SYSTEM_PROCESSOR`, `CMAKE_GENERATOR_PLATFORM` , spent three days dearching answer - no success.

The only working solution is here in this PR. This works allways. Doesn't break anything.